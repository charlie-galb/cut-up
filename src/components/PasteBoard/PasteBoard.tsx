import React from "react"

import { DragOverlay } from "@dnd-kit/core"
import { SortableContext, rectSortingStrategy} from "@dnd-kit/sortable"

import { TextSnippet } from "../TextSnippet/TextSnippet"
import { SortableTextSnippet } from "../TextSnippet/SortableTextSnippet"
import { chunkContainer } from "../../types/chunkContainer"

interface Props {
    chunkContainer: chunkContainer
    wordChunks: {
        [key: string]: string
      }
    activeId: string
}

export const PasteBoard = (props: Props) => {
   
    const { chunkContainer, wordChunks, activeId } = props
    const { id, nestedChunkIDs } = chunkContainer

    return (
        <div> 
            <SortableContext id={id} items={nestedChunkIDs} strategy={rectSortingStrategy}>
                <div className="unused-snippets" data-testid="unused-snippets" >
                    {nestedChunkIDs?.map((ID, i) => {
                        return (
                            <SortableTextSnippet data-testid="snippet" key={ID} id={ID} index={i} text={wordChunks[ID]}/>
                        )
                    })}
                </div>
            </SortableContext>
            <DragOverlay>
                {activeId ? <TextSnippet /> : null}
            </DragOverlay>
        </div>
    )
}
