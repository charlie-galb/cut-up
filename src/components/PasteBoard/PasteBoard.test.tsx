import React from "react"
import { render } from "@testing-library/react"
 
import { PasteBoard } from "./PasteBoard"

const wordChunks = ["chunk one", "chunk two"]

describe("PasteBoard", () => {
    test("It renders correctly", () => {
        const board = render(<PasteBoard wordChunks={wordChunks}/>)
        expect(board).toMatchSnapshot()
    })
    test("It displays text chunks as draggable objects", () => {
        const { getAllByRole } = render(<PasteBoard wordChunks={wordChunks}/>)
        const listItems = getAllByRole('listitem')
        expect(listItems).toHaveLength(2)
    })
})