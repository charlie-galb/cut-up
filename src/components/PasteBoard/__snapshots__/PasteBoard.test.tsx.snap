// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasteBoard It displays the correct number of text snippets 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pasteboard-container"
      >
        <div>
          <div
            class="unused-snippets"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="chunk-container-1"
            data-testid="pasteboard"
          >
            <div
              aria-describedby="rbd-hidden-text-0-hidden-text-0"
              class="text-snippet"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="snippet1"
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="snippet1"
              draggable="false"
              role="button"
              tabindex="0"
            >
              one and
            </div>
            <div
              aria-describedby="rbd-hidden-text-0-hidden-text-0"
              class="text-snippet"
              data-rbd-drag-handle-context-id="0"
              data-rbd-drag-handle-draggable-id="snippet2"
              data-rbd-draggable-context-id="0"
              data-rbd-draggable-id="snippet2"
              draggable="false"
              role="button"
              tabindex="0"
            >
              two and
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-0-hidden-text-0"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="pasteboard-container"
    >
      <div>
        <div
          class="unused-snippets"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="chunk-container-1"
          data-testid="pasteboard"
        >
          <div
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            class="text-snippet"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="snippet1"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="snippet1"
            draggable="false"
            role="button"
            tabindex="0"
          >
            one and
          </div>
          <div
            aria-describedby="rbd-hidden-text-0-hidden-text-0"
            class="text-snippet"
            data-rbd-drag-handle-context-id="0"
            data-rbd-drag-handle-draggable-id="snippet2"
            data-rbd-draggable-context-id="0"
            data-rbd-draggable-id="snippet2"
            draggable="false"
            role="button"
            tabindex="0"
          >
            two and
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
